src/index.js

```diff
@@ -8,26 +8,31 @@ const options = Object.freeze({
 
 const NightModeToggle = ({ size, checked, onChange, ...extraProps }) => {
   const ref = useRef();
-  const [progress, setProgress] = useState(() => 0);
+  const [progress, setProgress] = useReducer((_,next) => next, 0);
```

Comment:
What does `useReducer` buy us in this instance, if we're not sending actions? Again, just out of interest!